ckanext/resource_visibility/utils.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+
+import uuid
+import json
+import logging
+from datetime import datetime as dt
+
+import ckan.plugins.toolkit as tk
+from ckan.lib.mailer import mail_recipient, MailerException
+
+import ckanext.resource_visibility.constants as const
+from ckanext.resource_visibility.helpers import has_user_permission_for_org
+
+logger = logging.getLogger(__name__)
+
+
+def user_is_editor_or_admin(context, data_dict):
+    org_id = data_dict['owner_org']
+    user = context['auth_user_obj']
+    return has_user_permission_for_org(org_id, user, 'create_dataset')
+
+
+def save_updated_privacy_assessment_result(res_data):
+    pkg_dict = tk.get_action("package_show")({
+        "ignore_auth": True
+    }, {
+        "id": res_data["package_id"]
+    })
+
+    res_url = tk.h.url_for("resource.read",
+                           id=res_data["package_id"],
+                           resource_id=res_data["id"],
+                           _external=True)
+    data_dict = {
+        "id": res_data["id"],
+        "package_id": res_data["package_id"],
+        "maintainer": pkg_dict["maintainer"].strip(),
+        const.FIELD_ASSESS_RESULT: res_data[const.FIELD_ASSESS_RESULT],
+        "url": res_url,
+    }
+
+    update_upd_assessment_result_data(data_dict)
+
+
+def get_updated_privacy_assessment_result():
+    task = _get_upd_assessment_result_task()
+
+    return json.loads(task['value'])
+
+
+def update_upd_assessment_result_data(resource):
```

Comment:
What is "upd"?

Please don't tell me this is "update updated data". That's just :s